pkg/ottl/ottlfuncs/func_murmurhash3.go

```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"encoding/binary"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/spaolacci/murmur3"
+)
+
+const (
+	v32  = "32"
+	v128 = "128" // default
+)
+
+type MurmurHash3Arguments[K any] struct {
+	Target  ottl.StringGetter[K]
+	Version ottl.Optional[string] // 32-bit or 128-bit
+}
+
+func NewMurmurHash3Factory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("MurmurHash3", &MurmurHash3Arguments[K]{}, createMurmurHash3Function[K])
+}
+
+func createMurmurHash3Function[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*MurmurHash3Arguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("MurmurHash3Factory args must be of type *MurmurHash3Arguments[K]")
+	}
+
+	version := v128
+	if !args.Version.IsEmpty() {
+		if (args.Version.Get() != v32) && (args.Version.Get() != v128) {
+			return nil, fmt.Errorf("invalid arguments: %s. Version should be either \"32\" or \"128\"", args.Version.Get())
+		}
+		version = args.Version.Get()
+	}
+
+	return MurmurHash3HexString(args.Target, version)
+}
+
+// MurmurHash3HexString returns the hexadecimal representation of the hash in little-endian format.
+// MurmurHash3, developed by Austin Appleby, is sensitive to endianness. Unlike some other languages like Python,
+// which use little-endian for all architectures, the Go library `spaolacci/murmur3` has some open issues
```

Comment:
It might be helpful to directly reference the open issues being referenced. That way in the future we could remove this if they're ever fixed, and just to be able to quickly see the underlying issue.